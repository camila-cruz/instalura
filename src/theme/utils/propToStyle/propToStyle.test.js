import { propToStyle } from '.';

describe('propToStyle()', () => {
  describe('when it receives a simple argument', () => {
    test('and it is a string', () => {
      const propToStyleResult = propToStyle('textAlign');

      // <Text textAlign="center" />
      const componentProps = { textAlign: 'center' }; // number || string

      const styleResult = propToStyleResult(componentProps);

      expect(styleResult).toEqual({ textAlign: 'center' });
    });
    test('and it is a number', () => {
      const propToStyleResult = propToStyle('flex');

      // <Text textAlign={1} />
      const componentProps = { flex: 1 }; // number || string

      const styleResult = propToStyleResult(componentProps);

      expect(styleResult).toEqual({ flex: 1 });
    });
  });

  describe('when it receives an argument with breakpoints', () => {
    test('it renders one breakpoint resolution', () => {
      const propToStyleResult = propToStyle('textAlign');

      // <Text textAlign="center" />
      const componentProps = { textAlign: { xs: 'center' } };

      const styleResult = propToStyleResult(componentProps);

      expect(styleResult).toMatchSnapshot();
    });
    test('it renders two or more breakpoints resolutions', () => {
      const propToStyleResult = propToStyle('textAlign');

      // <Text textAlign="center" />
      const componentProps = { textAlign: { xs: 'center', md: 'right' } };

      const styleResult = propToStyleResult(componentProps);

      expect(styleResult).toMatchSnapshot();
    });
  });
});
