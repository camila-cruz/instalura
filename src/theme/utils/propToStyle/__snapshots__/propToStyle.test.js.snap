// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`propToStyle() when it receives an argument with breakpoints it renders one breakpoint resolution 1`] = `
Array [
  Array [
    "@media screen and (min-width:",
    "0",
    "px){",
    "text-align: center;",
    "}",
  ],
]
`;

exports[`propToStyle() when it receives an argument with breakpoints it renders two or more breakpoints resolutions 1`] = `
Array [
  Array [
    "@media screen and (min-width:",
    "0",
    "px){",
    "text-align: center;",
    "}",
  ],
  Array [
    "@media screen and (min-width:",
    "768",
    "px){",
    "text-align: right;",
    "}",
  ],
]
`;
